libsolidity/interface/CompilerStack.cpp

```diff
@@ -594,8 +594,13 @@ bool CompilerStack::checkLibraryNameClashes()
 string CompilerStack::absolutePath(string const& _path, string const& _reference) const
 {
 	// Anything that does not start with `.` is an absolute path.
-	if (_path.empty() || _path.front() != '.')
+	size_t slashPos { _path.find_first_of('/') };
```

Comment:
I think it is better to consistently use `boost::filesystem::path` throughout this function, i.e. first parse the whole path into a `boost::filesystem::path` (as done in line 600 / 605) and then just check for `p[0] == "." || p[0] == ".."`.